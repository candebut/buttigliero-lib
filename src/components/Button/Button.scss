// $colors: (
//     'PRIMARY_TURQOISE' : '#4CCFE2',
//     'PRIMARY_TURQOISE_HOVER' : '#40BED7',
//     'PRIMARY_PURPLE' : '#93A2DD',
//     'PRIMARY_PURPLE_HOVER' : '#8B99CF',
//     'PRIMARY_BLUE' : '#5D78FF',
//     'PRIMARY_BLUE_HOVER' : '#586BB5',
    
//     // LIGHT COLORS
//     'LIGHT_TURQOISE' : '#DFF9FD',
//     'LIGHT_ORANGE' : '#FFECC3',
//     'LIGHT_RED' : '#FDDFE8',
//     'LIGHT_GREEN' : '#DFFDE6',
//     'LIGHT_GRASS_GREEN' : '#EEFDDF',
//     'LIGHT_GREY' : '#EBEDF2',
//     'LIGHT_ICE_BLUE' : '#F5FAFF',
//     'LIGHT_CHARCOAL_GREY' : '#F0F1F5',
//     'LIGHT_BLUE_GREY' : '#EBF1F6',
    
//     // VIVID COLORS
//     'ORANGE' : '#F7941D',
//     'RED' : '#FF3061',
//     'GREEN' : '#99BE6A',
//     'YELLOW' : '#FFB822',
    
//     // MID/DARK COLORS
//      'ALMOST_BLACK' '#212529',
//      'DARK_GREY' '#80849B',
//      'MID_GREY' '#B2B9CD',
//      'DARK_BLUE' '#364C8D',
//      'DARK_GREEN' '#27B34B',
//      'DARK_TURQOISE' '#028698',
    
//     // NEUTRAL
//     'WHITE' '#ffffff',
//      'BLACK' '#000000',
    
//     // NOT NAMED YET
//      'LIGHTER_BLUE' '#A2C0DF',
//      'VERY_LIGHT_GREY' '#fafcff',
//      'LIGHTER_GREY' '#b1b3ba',
//      'LIGHTEST_GREY' '#f4f5f8',
//      'DARKER_GREY' '#9093A7',
//      'DARKEST_GREY' '#575962',
//      'ALTERNATIVE_BLUE' '#387FC9',
// );

$PRIMARY_TURQOISE: '#4CCFE2';
$PRIMARY_PURPLE: '#93A2DD';




.button-wrapper {
    position: relative;
    display: inline-block;
    width: auto;

    .button-component {
    font-family: Poppins;
    border-radius: 3px;
    display: flex;
    align-items: center;
    cursor: pointer;
    box-sizing: border-box;
    transition: all 200ms;

    &-primary {
        color: $PRIMARY_TURQOISE
    };
    &-secundary {
        color: $PRIMARY_PURPLE
    };
    }

}