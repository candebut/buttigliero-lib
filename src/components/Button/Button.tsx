import React from "react";
import { constants } from "../../constants/constants";
import "../../styles/scss/_button.scss";

export interface ButtonProps {
  label: string;
  variant: string;
  onClick?: React.MouseEventHandler;
}

const Button = ({ label, variant = constants.primary }: ButtonProps) => {
  const onClick = () => {
    if (onClick) {
      onClick();
    }
  };
  return (
    <div className="button-wrapper">
      <button className={`button-component__${variant}`}>{label}</button>
    </div>
  );
};

export default Button;
