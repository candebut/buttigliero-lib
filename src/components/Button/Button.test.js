import React from 'react';
import Button from './Button';
import { shallow, mount } from 'enzyme';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Button: snapshot tests', () => {
  it('should render correctly in primary variant', () => {
    const component = shallow(<Button variant="primary">Test</Button>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly in secundary variant', () => {
    const component = shallow(<Button variant="secundary">Test</Button>);
    expect(component).toMatchSnapshot();
  });
});

test('renders the button', async () => {
  render(<Button />);
  expect(screen.getByTestId('button-component')).toBeInTheDocument();
});
