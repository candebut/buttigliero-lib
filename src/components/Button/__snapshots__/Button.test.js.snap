// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: snapshot tests should render correctly in primary variant 1`] = `
<div
  className="buttonWrapper"
>
  <button
    className="button-component primary"
  />
</div>
`;

exports[`Button: snapshot tests should render correctly in secundary variant 1`] = `
<div
  className="buttonWrapper"
>
  <button
    className="button-component secundary"
  />
</div>
`;
